src/internal/ErrorBoundary/index.tsx

```diff
@@ -31,6 +32,12 @@ export default class ErrorBoundary extends React.Component<
 		return { error };
 	}
 
+	componentDidUpdate(prevProps: any) {
+		if (!isEqual(prevProps, this.props)) {
```

Comment:
Deep comparisons are not really recommended, as they can get pretty slow, and props shouldn't be mutated anyway. If you still think you have to keep this, then check if `error` is truth first before doing the deep comparison, so we limit it to only when there is an actual error.

Reply:
Yeah fair point. I _think_ this should be ok with a shallow comparison. I'll add the check in for `this.state.error` as well as an additional efficiency.